docs/migrations/web/MIGRATION-v3.md

```diff
@@ -6,9 +6,163 @@ Introducing version 3 of the _spirit-web_ package.
+
+### Helpers: Typography `text` infix and Heading emphasis
+
+As the word `text` was omitted from the typography tokens, the typography helpers
+were updated to reflect this change. Also, the `heading` helpers were updated to
+support the emphasis.
+
+#### Migration Guide
+
+From `body` typography helpers remove the `text` infix.
+
+For example:
+
+- `.typography-body-large-text-bold` → `.typography-body-large-bold`
+- `.typography-body-xsmall-text-regular` → `.typography-body-xsmall-regular`
+
+Rename `text` to `bold` in the `heading` helpers.
+
+For example:
+
+- `.typography-heading-xlarge-text` → `.typography-heading-xlarge-bold`
+- `.typography-heading-xsmall-text` → `.typography-heading-xsmall-bold`
+
+### Utilities: Removed and new classes
+
+Some utility classes were removed and new ones were added.
+
+Removed utility classes:
+
+- `.text-brand-primary`
+- `.text-brand-secondary`
+- `.bg-basic`
+- `.bg-cover`
+- `.bg-inverted`
+
+New utility classes:
+
+- `.text-tertiary`
+- `.bg-primary`
+- `.bg-secondary`
+- `.bg-tertiary`
+
+### Migrating product components
+
+First, you need to update your tokens. You can either use Supernova to export
+your tokens in new structure, or you can manually update your tokens or update
+our tokens package if you are using it.
+
+Here are some steps to migrate your product components to use the new tokens and themes:
+
+1. If you load `foundation`, `components`, `helpers`, and `utilities` separately, you need
+   add `@forward 'themes';` before them in your main file. If you load whole `scss` folder,
+   you don't need to do anything.
+2. Replace module load `@use '@tokens' as tokens;` in your components with `@use '@global' as global-tokens;`.
+3. Rename all the tokens in your components to use the new structure.
+   1. Replace all module name `tokens.*` → `global-tokens.*`.
+   2. Replace all spacing tokens with new values. See [Spacing values](#spacing-values) for more information.
+   3. Replace all radius tokens with new values. We added new radius token in between, so you need to update them.
+   4. Replace all typography tokens with new values. We removed `text` infix from typography tokens and added emphasized headings.
+   5. Switch all color tokens to use CSS variables. You can use `globals` settings module to load prefix.
+   6. If you used `$border-style` tokens, you need to replace them with direct values.
+   7. If you used `$focus` token, you need to replace it with `$focus-ring` token.
+   8. If you used `$grid-gutter` tokens, you need to replace them with `$grid-spacing` tokens.
+4. That's it! You should now have your components updated to use the new tokens and themes.
+
+Here is a simple example of how to update your component styles:
+
+Before:
+
+```scss
+@use '@tokens' as tokens;
+
+$color: tokens.$text-primary;
+$gap: tokens.$space-400;
+```
+
+After:
+
+```scss
+@use '@global' as global-tokens;
+@use '@lmc-eu/spirit-web/scss/settings/globals' as globals;
```

Comment:
The two "globals" here are a bit confusing, we should think about it again…

This is the minimum, I think:

```suggestion
@use '@global' as global-tokens;
@use '@lmc-eu/spirit-web/scss/settings/globals' as global-settings;
```

BTW, do we want the product to use _our_ globals at all? Shouldn't they apply their own prefix?

Reply:
1. Maybe we should rename the `@global` file as we already started to discuss. Maybe `@global-tokens` can be an option.
2. We either want them to use our global-settings or we should do the token prefix more properly. The problem is that settings are in the web package, but we already set the `spirit` prefix in the exporter and therefore in the `design-tokens` package. I am talking about shadows and gradients. Maybe if we make the prefix a token and use it everywhere from there, we can make this work and make it configurable. But this might not be easy.